### PR TITLE
Automate the TODO format check and fix older format

### LINKED ISSUE #2178: [Code health] All TODOs should be actionable and backed by tracking bugs in GitHub or removed

### LINKED ISSUE #2841: [Code health] Replace bug #s in code with full URLs
..to make issues more discoverable and easier to access via a single click.

### PR DESCRIPTION
**Pull request by anandwana001**
Fixes #2178
Fixes #2841

@gino-m  @shobhitagarwal1612  PTAL?

### CODE REVIEW

**Review by gino-m — CHANGES_REQUESTED** [review]
Thanks @anandwana001 for adding the detekt check. A single ticket isn't really helpful for tracking those TODO, though - ideally we'd have a ticket for each issue or class of issues.

Also note that I've started adding the full URL of the ticket to the `TODO()` tag so that they're clickable in Android Studio. That will usually cause the TODO to wrap onto the next line, in which case you can intent two spaces to ensure AS interprets the second line as part of the same TODO.

**Comment by anandwana001** [comment]
@gino-m 
The goal of this PR:

1. To enable the TODO format check
2. All the untracked TODO, will be connected with a single issue #2903, where any developer in the future can create sub-tickets and work on it.
3. As per the PR, there are 38 untracked issues, for which we have to create individual tickets with all details, might take time, and block us from enabling the TODO format check

Hence this PR. 

Updating the detekt check as per the suggestion for a clickable link

**Review by gino-m — CHANGES_REQUESTED** [review]
I've added links to appropriate issues, creating them where appropriate. 

Idea: Can we use https://github.com/marketplace/actions/todo-to-issue to auto-generate issues for each TODO as they're added in the future? In that case we'd have a different format for TODO URLs:

```
// TODO: Todo description
// Issue URL: ...github url.
```

@shobhitagarwal1612 @scolsen Wdyt? If you agree I can enable the plugin and try it out.
Wdyt? If you like the idea I can enable the plugin for you to try out.

**Comment by gino-m** [comment]
We spoke. 
- [ ] @shobhitagarwal1612 will set up the GitHub action to create issues on merge to `master`.
- [ ] @anandwana001 Please move URLs to 2nd line as "Issue URL" (see above)
- [ ] @anandwana001 Also update check to require TODOs to look like `// TODO: ...`

File: config/detekt/detekt.yml [hunk 1]
```diff
@@ -590,9 +590,9 @@ style:
                 value: 'FIXME:'
             -   reason: 'Forbidden STOPSHIP todo marker in comment, please address the problem before shipping the code.'
                 value: 'STOPSHIP:'
-            -   reason: 'TODO comments must reference a GitHub issue for tracking. Format: TODO(https://github.com/google/ground-android/issues/IssueNumber).'
-                value: '(TODO|TODO\(|TODO\(#)'
-        allowedPatterns: '^TODO\(https://github\.com/google/ground-android/issues/\d+\): .+'
+            -   reason: 'TODO comments must reference a GitHub issue for tracking. Format: TODO: and Issue URL: in next line.'
+                value: 'TODO:'
+        allowedPatterns: '^//\s*TODO:\s+.+'
```
Comment (gino-m) [inline config/detekt/detekt.yml:595]: There should be exactly one before, and one space after "TODO", so `^//.TODO:/.` Multiple arbitrary whitespace shouldn't be allowed (e.g., "~//&nbsp;&nbsp;&nbsp;&nbsp;TODO:&nbsp;&nbsp;&nbsp;Do something~")

File: ground/build.gradle [hunk 2]
```diff
@@ -358,7 +359,8 @@ dependencies {
 
     implementation "com.google.guava:guava:33.0.0-android"
 
-    // TODO(https://github.com/google/ground-android/issues/1748): Move protos into shared module and set correct path here.
+    // TODO: Move protos into shared module and set correct path here.
+    // Issue URL: https://github.com/google/ground-android/issues/1748
```
Comment (gino-m) [inline ground/build.gradle:363]: Thanks for making these changes

**Review by gino-m — CHANGES_REQUESTED** [review]
Once nit, otherwise LGTM

**Review by gino-m — APPROVED** [review]

File: config/detekt/detekt.yml [hunk 3]
```diff
@@ -590,9 +590,9 @@ style:
                 value: 'FIXME:'
             -   reason: 'Forbidden STOPSHIP todo marker in comment, please address the problem before shipping the code.'
                 value: 'STOPSHIP:'
-            # -   reason: 'Forbidden TODO todo marker in comment, please do the changes.'
-            #    value: 'TODO:'
-        allowedPatterns: ''
+            -   reason: 'TODO comment format should be: // TODO: $description'
+                value: '// TODO:'
+        allowedPatterns: '^//\s*TODO:\s+.+$'
```
Comment (gino-m) [inline config/detekt/detekt.yml:595]: Sorry my original suggestion wasn't correct, but this also seems like the original regex. I think you want this:

`allowedPatterns: '^// TODO: .+$'`

**Review by gino-m — CHANGES_REQUESTED** [review]

**Review by gino-m — APPROVED** [review]

**Pull request merged**